### PR TITLE
Bound by domain

### PR DESCRIPTION
**Pull request by farkaskid**

### CODE REVIEW

**Comment by mahesh1996** [comment]
What about bounding by hostname. Current implementation matches domain anywhere inside URL.

**Comment by mahesh1996** [comment]
https://golang.org/pkg/net/url/#Parse

**Comment by farkaskid** [comment]
Revisiting the implementation. We won't need the `-bound` to be a string flag. It can just be a boolean flag.

**Comment by mahesh1996** [comment]
Then if I want to list  urls having host name xyz in url abc, it won't wotk.
E.g. I want to list all netflix urls to which google is pointing.

**Comment by farkaskid** [comment]
🤔 this is an useful functionality but it seem like the responsibility of the `collector` logic.
We can keep the `bound` flag and use a different collector that just collects collects netflix's URLs.
When `bound` flag is active, URLs from other domains(like Netflix) that are pointed by URLs in the given domain(like Google) are collected but are **not** visited.

They are currently being filtered by the `CrossDomainFilter`.

 So we can collect all the Netflix URLs that Google is pointing but it requires a slightly tweaked collector **along** with the `bound` flag.

We can discuss this in detail and add a feature request issue for this. We will have to expose configuration for choosing different `collectors` to achieve this.

File: README.md [hunk 1]
```diff
@@ -7,3 +7,8 @@ A simple web crawler. This leverages Go's lightweight goroutines to achieve high
 and enjoy.
 
 OSX binary is available in the release.
```
Comment (mahesh1996) [inline README.md]: Any further info on How user can use this binary?
Any steps needed?

File: crawler/filter.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	"net/url"
+)
+
+type NoneFilter struct{}
+
+func (filter NoneFilter) Filter(urls []string) []string {
+	return urls
+}
+
+type CrossDomainFilter struct {
```
Comment (mahesh1996) [inline crawler/filter.go:13]: Should we move types into a specific location and function into other file

File: main.go [hunk 3]
```diff
@@ -4,20 +4,40 @@ import (
...
+		log.Fatalln(err)
+	}
+
+	log.Println(*bound)
+
+	if *bound {
+		log.Println(url.Hostname())
+		filter = crawler.CrossDomainFilter{url.Hostname()}
+	} else {
+		filter = crawler.NoneFilter{}
```
Comment (mahesh1996) [inline main.go:31]: Does go have DI inbuilt or any framework?
😆

**Review by mahesh1996 — CHANGES_REQUESTED** [review]

File: main.go (hunk 3 above)
Comment (farkaskid) [reply main.go:31]: We can add a static DI frameworks but they are not needed for the filters and they are only made once.

File: README.md (hunk 1 above)
Comment (farkaskid) [reply README.md]: Altho, downloading and running a binary should be obvious to any dev but still added that you can use the `./crawlerOSX` syntax for assistance.

File: crawler/filter.go (hunk 2 above)
Comment (farkaskid) [reply crawler/filter.go:13]: discussed.

**Review by mahesh1996 — APPROVED** [review]

**Pull request merged**